roles/puppet_agent/defaults/main.yml

```diff
@@ -0,0 +1,12 @@
+---
+puppet_package_name: puppet-agent
+puppet_package_state: installed
+puppet_package_manage: true
+
+puppet_group_name: puppet
+puppet_group_state: present
+puppet_group_manage: true
```

Comment:
What situations lead to not wanting this group managed?

Reply:
Again, I like things to be configurable. It's from my Puppet experience: configurable but a sane default.

Reply:
The downside is over specifying configuration. In part my questions are to make sure we focus on specifying what users should be configuring, and usings `vars` to put things that are internally specified without hardcoding the values. I tend to also find that over specifying configuration leads to hard to read code. So in this case, I am asking myself, do people change the puppet_agent group? Do they ever want to not manage it? This stuff also seems like it tends to be rarely if ever changed and thus a sort of pre-optimization.